chasm/lib/callback/component.go

```diff
@@ -51,3 +59,72 @@ func (c *Callback) recordAttempt(ts time.Time) {
 	c.Attempt++
 	c.LastAttemptCompleteTime = timestamppb.New(ts)
 }
+
+//nolint:revive // context.Context is an input parameter for chasm.ReadComponent, not a function parameter
+func (c *Callback) loadInvocationArgs(
+	chasmCtx chasm.Context,
+	ctx context.Context,
+) (callbackInvokable, error) {
+	target, err := c.CompletionSource.Get(chasmCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO (seankane,yichao): we should be able to use the chasm context here.
+	completion, err := target.GetNexusCompletion(ctx, c.RequestId)
+	if err != nil {
+		return nil, err
+	}
+
+	switch variant := c.GetCallback().GetVariant().(type) {
+	case *callbackspb.Callback_Nexus_:
+		if variant.Nexus.Url == chasm.NexusCompletionHandlerURL {
+			return chasmInvocation{
+				nexus:      variant.Nexus,
+				attempt:    c.Attempt,
+				completion: completion,
+				requestID:  c.RequestId,
+			}, nil
+		}
+		return nexusInvocation{
+			nexus:      variant.Nexus,
+			completion: completion,
+			// workflowID: c.WorkflowId,
+			workflowID: chasmCtx.ExecutionKey().BusinessID,
+			runID:      chasmCtx.ExecutionKey().EntityID,
+			attempt:    c.Attempt,
+		}, nil
+	default:
+		return nil, queues.NewUnprocessableTaskError(
+			fmt.Sprintf("unprocessable callback variant: %v", variant),
+		)
+	}
+}
+
+func (c *Callback) saveResult(
+	ctx chasm.MutableContext,
+	result invocationResult,
+) (struct{}, error) {
```

Comment:
Not blocking this PR but we need to add a `chasm.NoValue` type that is `*struct{}` and just return `nil` from methods that don't have a return value.

Reply:
#8566

Reply:
I added this type FTR.